hazelcast/src/main/java/com/hazelcast/queue/impl/QueueService.java

```diff
@@ -299,4 +301,15 @@ public void rollbackTransaction(String transactionId) {
             operationService.executeOperation(operation);
         }
     }
+
+    @Override
+    public Map<String, LocalQueueStats> getStats() {
+        Map<String, LocalQueueStats> queueStats = MapUtil.createHashMap(containerMap.size());
+        for (Entry<String, QueueContainer> queueStat : containerMap.entrySet()) {
+            queueStats.put(queueStat.getKey(), createLocalQueueStats(queueStat.getKey(),
```

Comment:
I fully agree. This is extremely hard to understand code. It took me 5 minutes to untangle even though the intent of the code is simple.
Round 1:

```
@Override
public Map<String, LocalQueueStats> getStats() {
	Map<String, LocalQueueStats> queueStats = MapUtil.createHashMap(containerMap.size());
	for (Entry<String, QueueContainer> queueStat : containerMap.entrySet()) {
		queueStats.put(
			queueStat.getKey(), 
			createLocalQueueStats(
				queueStat.getKey(),
				nodeEngine.getPartitionService().getPartitionId(
					nodeEngine.getSerializationService().toData(
						queueStat.getKey(), 
						StringPartitioningStrategy.INSTANCE))));
	}
	return queueStats;	
}
```
See the amount of nesting of data!

Round 2:
```
@Override
public Map<String, LocalQueueStats> getStats() {
	Map<String, LocalQueueStats> queueStats = MapUtil.createHashMap(containerMap.size());
	SerializationService serializationService = nodeEngine.getSerializationService()	
	PartitionService partitionService = nodeEngine.getPartitionService();	
	for (Entry<String, QueueContainer> queueStat : containerMap.entrySet()) {
		String key = queueStat.getKey();
		Data keyData = serializationService.toData(key, StringPartitioningStrategy.INSTANCE)
		int partitionId = partitionService.getPartitionId(keyData);
		LocalQueueStatus value= createLocalQueueStats(partitionId);		
		queueStats.put(key,value);
	}
	return queueStats;	
}
```

It can still be cleaned up because often you want to pull out the content of a for loop:

```
@Override
public Map<String, LocalQueueStats> getStats() {
	Map<String, LocalQueueStats> result = createHashMap(containerMap.size());
	for (Entry<String, QueueContainer> entry : containerMap.entrySet()) {
		LocalQueueStats stats = createLocalQueueStats(entry.getKey());		
		result.put(key,stats);
	}
	return result;	
}
```

And unless I messed up the code it can even be simplified to:
@Override
public Map<String, LocalQueueStats> getStats() {
	Map<String, LocalQueueStats> result = createHashMap(containerMap.size());
	for (String key : containerMap.keySet()) {
		LocalQueueStats stats = createLocalQueueStats(key);		
		result.put(key,stats);
	}
	return result;	
}
```

This is super to read code